docs/adminguide/hosting/setup-tls.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# cspell:words keyout noenc CAcreateserial
+
+function generate_cert_conf() {
+    local name=$1
+    local san=$2
+
+    cat << EOF > $name.crt.conf
+[req]
+distinguished_name = req_dist_name
+req_extensions = req_ext
+prompt = no
+
+[req_dist_name]
+CN = $name
+
+[req_ext]
+subjectAltName = $san
+EOF
+}
+
+function generate_certificate_request() {
+    local name=$1
+    openssl req -batch \
+        -new -sha512 -noenc -newkey rsa:4096 \
+        -config $name.crt.conf \
+        -keyout $name.key -out $name.csr
+}
+
+function sign_crt_with_ca() {
+    local ca_crt=$1
+    local ca_key=$2
+    local name=$3
+
+    openssl x509 -req -in $name.csr \
+        -CA $ca_crt -CAkey $ca_key \
+        -extfile $name.crt.conf \
+        -extensions req_ext \
+        -CAcreateserial -out $name.crt \
+        -days 10 -sha512
+}
+
+if [ ! -f custom-ca.key ]; then
+    echo "Generate a mini Certificate Authority"
+    openssl req -batch \
+        -x509 -sha512 -nodes -days 10 -newkey rsa:4096 \
+        -subj "/CN=Mini Certificate Authority" \
+        -keyout custom-ca.key -out custom-ca.crt
+fi
+
+for service in parsec-{s3,backend}; do
+    if [ ! -f $service.crt.conf ]; then
+        echo "Generating conf for $service"
+        generate_cert_conf $service DNS:$service,DNS:localhost,IP:127.0.0.1
+    fi
+
+    if [ ! -f $service.key ]; then
+        echo "Generate certificate request for $service"
```

Comment:
```suggestion
        echo "Generate certificate request ${service}.csr"
```

Reply:
done, moved to `generate_certificate_request`